app/controllers/container_topology_controller.rb

```diff
@@ -0,0 +1,105 @@
+class ContainerTopologyController < ApplicationController
+  before_action :check_privileges
+  before_action :get_session_data
+  after_action :cleanup_action
+  after_action :set_session_data
+
+  def show
+    topology = build_topology
+    @topologyitems = topology[:items].to_json
+    @topologyrelations = topology[:relations].to_json
+    @topologykinds = topology[:kinds].to_json
+  end
+
+  def index
+    redirect_to :action => 'show'
+  end
+
+  def data
+    render :json => {:data => build_topology}
+  end
+
+  private
+
+  def get_session_data
+    @layout = "container_topology"
+  end
+
+  def set_session_data
+    session[:layout] = @layout
+  end
+
+  def build_topology
+    nodes, services = entities
+    topology = {}
+    topo_items = {}
+    links = []
+
+    nodes.each do |n|
+      topo_items[n.ems_ref] = {:metadata => {:id  => n.ems_ref, :name => n.name}, :kind => "Node"}
+      if n.container_groups.size > 0
```

Comment:
We should use Rubyisms here like `if n.container_groups.any?`

Actually...on second review this line isn't needed at all...when you have no items, .each is a noop, so let's just remove this.